topics/docker.md

```diff
@@ -1,270 +1,100 @@
-```
+* What image is going to be used (JDK 8 in this case).
+* The exposed port (this does not automatically expose the port which is done when running the container)
+* How to run the application (`cmd` file)
 
-The last line instructs Docker to run `java` with G10s GC, 4G of memory and your packaged application. 
+The other steps merely create a folder, copy the contents from the build output to the folder and change to it in preparation
+to run the image.
 
 ## Building and running the Docker image
 
-Build an application package:
+First step is to build the actual application (in this case using Gradle)
 
 ```bash
 ./gradlew build
 ```
 
-Build and tag an image:
+Next step is to build and tag the Docker image:
 
 ```bash
 docker build -t my-application .
 ```
 
-Start an image:
+Finally, start the image:
 
 ```bash
-docker run -m512M --cpus 2 -it -p 8080:8080 --rm my-application
+docker run -p 8080:8080 my-application
 ```
 
-With this command, we start Docker in a foreground mode. It will wait for the server to exit, it
-will also respond to `Ctrl+C` to stop it. `-it` instructs Docker to allocate a terminal (*tty*) to pipe the stdout
-and to respond to the interrupt key sequence. 
-
-Since our server is running in an isolated container now, we should tell Docker to expose a port so we can
-actually access the server port. Parameter `-p 8080:8080` tells Docker to publish port 8080 from inside a container as a port 8080 on a local
-machine. Thus, when you tell your browser to visit `localhost:8080` it will first reach out to Docker, and it will bridge
-it into internal port `8080` for your application. 
-
-You can adjust memory with `-m512M` and number of exposed cpus with `--cpus 2`. 
-
-By default, a container’s file system persists even after the container exits, so we supply `--rm` option to prevent
-garbage piling up.
-
 For more information about running a docker image please consult [docker run](https://docs.docker.com/engine/reference/run) 
 documentation.
 
-## Pushing docker image 
-
-Once your application is running locally successfully, it might be a time to deploy it:
+If using [IntelliJ IDEA](https://www.jetbrains.com/idea), you can simply click `Run` in the Dockcerfile
+to perform these steps
 
-```bash
-docker tag my-application hub.example.com/docker/registry/tag
-docker push hub.example.com/docker/registry/tag
-```
- 
-These commands will tag your application for a registry and push an image. 
-Of course, you need to replace `hub.example.com/docker/registry/tag` with an actual URL for your registry.
-
-We won't go into details here since your configuration might require authentication, specific configuration options 
-and even special tools. Please consult your organization or cloud platform, or 
-check [docker push](https://docs.docker.com/engine/reference/commandline/push/) documentation.
+![Docker Run](run-docker.png)
```

Comment:
We can add the `width` attribute here to scale a retina image:
```
![Docker Run](run-docker.png){width="291"}
```
Now this image is displayed in a double size.

Reply:
Thanks. Btw, let's also make sure that all images have Alt text. Found a few missing.